packages/code-block/partials/clipboard-button/index.tsx

```diff
@@ -46,26 +50,34 @@ function ClipboardButton({
     return handleSuccess()
   }
 
-  // Handle errors from copying-to-clipboard and invoke callback
+  // Handle errors from copying-to-clipboard and update state
   function handleError(errorMessage) {
     // Enhancement - is there anywhere we can send this error for tracking?
     console.error(errorMessage)
-    setCopiedState(false)
-
-    // If an onCopyCallback was provided, call it
-    if (typeof onCopyCallback == 'function') {
-      onCopyCallback(false)
-    }
+    updateCopiedState(false)
   }
 
-  // Track the event, set state, and invoke callback
+  // Track the event and update state
   function handleSuccess() {
     analytics.trackCopy()
-    setCopiedState(true)
+    updateCopiedState(true)
+  }
+
+  /**
+   * Handle setting state, re-focusing the button, and invoking the copy
+   * callback. This should not be called when `copiedState` is being reset, only
+   * when there is a success or failure of the copy action.
+   */
+  function updateCopiedState(newState: boolean) {
+    // Update the local state variable
+    setCopiedState(newState)
+
+    // Re-focus the copy button
+    buttonRef?.current?.focus()
```

Comment:
```suggestion
    // Re-focus the copy button as it loses focus when we call `.select()` on the textarea element in `copyToClipboard`
    buttonRef?.current?.focus()
```

Reply:
Updated in https://github.com/hashicorp/react-components/pull/605/commits/9490b914e6daf6d23fee60979bed4b5e850b9f58! Also updated the `onCopyCallback` description in https://github.com/hashicorp/react-components/pull/605/commits/54863c94f759d7ce1dcf123d0379fef74bb2c17f, which I should've done in the last PR. 😁